backend/review/management/commands/precompute_pcr_views.py

```diff
@@ -0,0 +1,101 @@
+import logging
+
+from django.core.cache import cache
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.http import Http404
+from tqdm import tqdm
+
+from courses.models import Topic
+from review.models import CachedReviewResponse
+from review.views import manual_course_reviews
+
+
+def precompute_pcr_views(verbose=False, is_new_data=False):
+    if verbose:
+        print("Now precomputing PCR reviews.")
+
+    CachedReviewResponse.objects.all().update(expired=True)
+    responses = CachedReviewResponse.objects.all()
+    topic_set = {response.topic_id: response for response in responses}
+
+    objs_to_insert = []
+    objs_to_update = []
+    cache_deletes = set()
+    has_count = total_count = 0
+
+    # Iterate through all topics.
+    with transaction.atomic():
+        for topic in tqdm(
+            Topic.objects.all().select_related("most_recent").order_by("most_recent__semester"),
+            disable=not verbose,
+        ):
+            course_id_list, course_code_list = zip(*topic.courses.values_list("id", "full_code"))
+            topic_id = ".".join([str(id) for id in sorted(course_id_list)])
+            total_count += 1
+
+            if topic_id in topic_set:
+                try:
+                    has_count += 1
```

Comment:
nit: usually try to minimize the stuff inside the try block to what is absolutely going to throw an error (for debug/code readability reasons)